presto-function-namespace-managers/src/main/java/com/facebook/presto/functionNamespace/mysql/MySqlFunctionNamespaceManager.java

```diff
@@ -0,0 +1,337 @@
+    {
+        Optional<UserDefinedType> type = functionNamespaceDao.getUserDefinedType(typeName.getCatalogName(), typeName.getSchemaName(), typeName.getObjectName());
+        if (!type.isPresent()) {
+            throw new PrestoException(NOT_FOUND, format("Type %s not found", typeName));
+        }
+        return type.get();
+    }
+
+    @Override
+    protected Collection<SqlInvokedFunction> fetchFunctionsDirect(QualifiedObjectName functionName)
+    {
+        checkCatalog(functionName);
+        return functionNamespaceDao.getFunctions(
+                functionName.getCatalogName(),
+                functionName.getSchemaName(),
+                functionName.getObjectName());
+    }
+
+    @Override
+    protected FunctionMetadata fetchFunctionMetadataDirect(SqlFunctionHandle functionHandle)
+    {
+        checkCatalog(functionHandle);
+        Optional<SqlInvokedFunction> function = functionNamespaceDao.getFunction(hash(functionHandle.getFunctionId()), functionHandle.getFunctionId(), getLongVersion(functionHandle));
+        if (!function.isPresent()) {
+            throw new InvalidFunctionHandleException(functionHandle);
+        }
+        return sqlInvokedFunctionToMetadata(function.get());
+    }
+
+    @Override
+    protected ScalarFunctionImplementation fetchFunctionImplementationDirect(SqlFunctionHandle functionHandle)
+    {
+        checkCatalog(functionHandle);
+        Optional<SqlInvokedFunction> function = functionNamespaceDao.getFunction(hash(functionHandle.getFunctionId()), functionHandle.getFunctionId(), getLongVersion(functionHandle));
+        if (!function.isPresent()) {
+            throw new InvalidFunctionHandleException(functionHandle);
+        }
+        return sqlInvokedFunctionToImplementation(function.get());
+    }
+
+    @Override
+    public void createFunction(SqlInvokedFunction function, boolean replace)
+    {
+        checkCatalog(function);
+        checkFunctionLanguageSupported(function);
+        checkArgument(!function.hasVersion(), "function '%s' is already versioned", function);
+
+        QualifiedObjectName functionName = function.getFunctionId().getFunctionName();
+        checkFieldLength("Catalog name", functionName.getCatalogName(), MAX_CATALOG_NAME_LENGTH);
+        checkFieldLength("Schema name", functionName.getSchemaName(), MAX_SCHEMA_NAME_LENGTH);
+        if (!functionNamespaceDao.functionNamespaceExists(functionName.getCatalogName(), functionName.getSchemaName())) {
+            throw new PrestoException(NOT_FOUND, format("Function namespace not found: %s", functionName.getCatalogSchemaName()));
+        }
+        checkFieldLength("Function name", functionName.getObjectName(), MAX_FUNCTION_NAME_LENGTH);
+
+        if (function.getParameters().size() > MAX_PARAMETER_COUNT) {
+            throw new PrestoException(GENERIC_USER_ERROR, format("Function has more than %s parameters: %s", MAX_PARAMETER_COUNT, function.getParameters().size()));
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

This code uses &#39;%s&#39; to format int: MAX_PARAMETER_COUNT, &#39;size&#39; expression. This is a potential locale-sensitive handling issue. It might cause errors in the handling and processing of the statement at line: 175. Consider formatting this data with &#39;%d&#39; instead.